jetty-servlet/src/test/java/org/eclipse/jetty/servlet/ServletContextHandlerTest.java

```diff
@@ -1651,21 +1651,35 @@ public void testProgrammaticFilterServlet() throws Exception
         ServletHandler handler = new ServletHandler();
         _server.setHandler(context);
         context.setHandler(handler);
-        handler.addServletWithMapping(new ServletHolder(new TestServlet()), "/");
+        handler.addServletWithMapping(new ServletHolder(new TestServlet()
+        {
+            @Override
+            public void init() throws ServletException
+            {
+                handler.addFilterWithMapping(new FilterHolder(new MyFilter()), "/test/*", EnumSet.of(DispatcherType.REQUEST));
```

Comment:
Can we put in some comments about what is really being tested here, which is the various states in which the ServletContextHandler, ServletHandler can be in when a servlet/filter/listener is added.